src/runtime/schema.ts

```diff
@@ -50,9 +37,10 @@ export class Schema {
   // For convenience, primitive field types can be specified as {name: 'Type'}
   // in `fields`; the constructor will convert these to the correct schema form.
   // tslint:disable-next-line: no-any
-  constructor(names: string[], fields: Dictionary<any>, description?) {
+  constructor(names: string[], fields: Dictionary<any>, description?, refinement?) {
```

Comment:
given multiple optional fields it might be better to convert this to an options dictionary input